Source/WebKit/Shared/XR/PlatformXR.serialization.in

```diff
@@ -95,13 +100,26 @@ enum class PlatformXR::XRTargetRayMode : uint8_t {
     Vector<WebCore::FloatPoint> bounds;
 };
 
+#if PLATFORM(COCOA)
+[Nested, RValue] struct PlatformXR::FrameData::LayerSetupData {
+    std::array<std::array<uint16_t, 2>, 2> physicalSize;
+    std::array<WebCore::IntRect, 2> viewports;
+    std::array<std::span<const float>, 2> horizontalSamples;
+    std::span<const float> verticalSamples;
+    WebCore::IntSize screenSize;
+    std::array<uint16_t, 2> framebufferSize;
+    MachSendRight completionSyncEvent;
+};
+#endif
+
 [Nested, RValue] struct PlatformXR::FrameData::LayerData {
 #if PLATFORM(COCOA)
-    WebCore::IntSize framebufferSize;
+    std::optional<PlatformXR::FrameData::LayerSetupData> layerSetup;
+    uint64_t renderingFrameIndex;
```

Comment:
Why was this reordered?

Reply:
because its used as a HashKey and hash keys in WTF have a limit of 150 bytes which we exceeded. So its reordered in size largest -> smallest to reduce struct size due to struct padding